engine/src/main/java/io/camunda/zeebe/engine/EngineImpl.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine;
+
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.engine.api.Engine;
+import io.camunda.zeebe.engine.api.EngineContext;
+import io.camunda.zeebe.engine.api.ErrorHandlingContext;
+import io.camunda.zeebe.engine.api.ProcessingContext;
+import io.camunda.zeebe.engine.api.ProcessingResult;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecord;
+import io.camunda.zeebe.engine.state.EventApplier;
+import io.camunda.zeebe.engine.state.ZeebeDbState;
+import io.camunda.zeebe.engine.state.mutable.MutableZeebeState;
+import java.util.function.Function;
+
+public class EngineImpl implements Engine {
```

Comment:
:wrench: Just as a note not necessarily something you have to change right now. The `engine` interface name might be a bit misleading if we have different implementations, like for exporters (not sure whether this will happen) or for something like the hotbackup.